src/app/InteractionModelEngine.cpp

```diff
@@ -1638,6 +1638,21 @@ void InteractionModelEngine::ResumeSubscriptionsTimerCallback(System::Layer * ap
     auto * iterator = imEngine->mpSubscriptionResumptionStorage->IterateSubscriptions();
     while (iterator->Next(subscriptionInfo))
     {
+        // If subscription happens between reboot and this timer callback, it's already live and should skip resumption
+        if (Loop::Break == imEngine->mReadHandlers.ForEachActiveObject([&](ReadHandler * handler) {
+                SubscriptionId subscriptionId;
+                handler->GetSubscriptionId(subscriptionId);
+                if (subscriptionId == subscriptionInfo.mSubscriptionId)
```

Comment:
But aren't subscription IDs assigned by the publisher?  I don't think this works.  What we would need to do is install the new subscription, but also evict persisted subscriptions per our eviction algorithm if resources (handlers and / or paths) may subsequently be exhausted when combining the new subscriptions and persisted subscriptions.

Predicting that resources will be exhausted will be difficult, and the need to do this would prevent us reusing our existing eviction code.  Given that, it seems like the simplest approach is to install the persisted subscriptions immediately on server startup, but delay initial priming reports until after the initial min interval.

Reply:
> But aren't subscription IDs assigned by the publisher? 

Yes, but this code _is_ running on the publisher.

Reply:
> install the persisted subscriptions immediately on server startup, but delay initial priming reports until after the initial min interval.

That would be doable, but a lot more surgery on the ReadHandler state machine....

Reply:
In general usage, when subscription resumption scenarios arise, the subscribers would be oblivious of the reboot, and resumption will happen as normal. What this PR aims to do is to close the edge case.

That said I do see what you are saying about eviction, @msandstedt. I've filed an issue https://github.com/project-chip/connectedhomeip/issues/25579

Reply:
And the solution for #25579 may be to either add an argument or create a variant for `EnsureResourceForSubscription` that does not evict, and use that for the subscription resumption loop.

Reply:
I don't understand how it covers any edge case though.  Do we have logic that will cause in-bound subscription requests to associate to the same subscription ID in the publisher that the matching persisted, precursor subscription did?

Reply:
We don't, and that's not what this is trying to handle.

What this is trying to handle is that when a subscription is established we persist it.  So the set of persisted subscriptions, by the time we go to resume them, would include any subscriptions that have gotten established between startup and now.  And we don't want to include them in the set of things we are trying to resume, since they are working fine already.

Reply:
Oh, I completely misunderstood.  Thanks for the clarification.  #25579 covers my concern then.